bundles/org.openhab.binding.tibber/src/main/java/org/openhab/binding/tibber/internal/handler/TibberHandler.java

```diff
@@ -509,7 +505,38 @@ private void handleNewMessage() {
             updateChannel(CHANNEL_GROUP_LIVE, CHANNEL_CURRENT_2, value, "A");
             value = Utils.getJsonValue(jsonData, "currentL3");
             updateChannel(CHANNEL_GROUP_LIVE, CHANNEL_CURRENT_3, value, "A");
+
+            String consumption = Utils.getJsonValue(jsonData, "power");
+            String production = Utils.getJsonValue(jsonData, "powerProduction");
+            updateChannel(CHANNEL_GROUP_LIVE, CHANNEL_CONSUMPTION, consumption, "W");
+            updateChannel(CHANNEL_GROUP_LIVE, CHANNEL_PRODUCTION, production, "W");
+            updateConsumptionAndProductionChannel(consumption, production);
+        }
+    }
+
+    private void updateConsumptionAndProductionChannel(String consumption, String production) {
+        double consumptionValue = 0.0;
+        double productionValue = 0.0;
+        try {
+            if (consumption != null && !consumption.isBlank() && !consumption.equals(EMPTY_VALUE)
+                    && !consumption.equals(NULL_VALUE)) {
+                consumptionValue = Double.parseDouble(consumption);
+            }
+        } catch (NumberFormatException e) {
+            logger.error("Unable to parse consumption value: {}. Assuming 0.", consumption, e);
         }
+
+        try {
+            if (production != null && !production.isBlank() && !production.equals(EMPTY_VALUE)
+                    && !production.equals(NULL_VALUE)) {
+                productionValue = Double.parseDouble(production);
+            }
+        } catch (NumberFormatException e) {
+            logger.error("Unable to parse production value: {}. Assume 0.", production, e);
+        }
+
+        updateChannel(CHANNEL_GROUP_LIVE, CHANNEL_CONSUMPTION_AND_PRODUCTION,
+                String.valueOf(consumptionValue - productionValue), "W");
     }
```

Comment:
Might need spotless, but this is a small improvement null annotated, no warnings, no repeating.
```suggestion
            double consumptionValue = parseValueSafely(consumption, "consumption");
            double productionValue = parseValueSafely(production, "production");

            updateChannel(CHANNEL_GROUP_LIVE, CHANNEL_CONSUMPTION_AND_PRODUCTION,
                String.valueOf(consumptionValue - productionValue), "W");
        }
    }

    private double parseValueSafely(@Nullable String value, String valueType) {
        if (value == null || value.isBlank() || EMPTY_VALUE.equals(value) || NULL_VALUE.equals(value)) {
            return 0.0;
        }
        
        try {
            return Double.parseDouble(value);
        } catch (NumberFormatException e) {
            logger.error("Unable to parse {} value: {}. Assuming 0.", valueType, value, e);
            return 0.0;
        }
    }
```